### PR TITLE
Change javascript linters to eslint

### LINKED ISSUE #228: Change javascript linters to eshint
We must have all repos as equal as possible to Wordpress-seo, so as discussed with @moorscode , the Grunt check command should contain eshint like Wordpress-seo

this issue is the answer to : #227

### PR DESCRIPTION
**Pull request by dutchmartin**
## Summary

This PR can be summarized in the following changelog entry:

* Removes the jshint and jsvalidate JavaScript checkers in favor of eslint.

## Relevant technical choices:

* Removing the jshint and jsvalidate JavaScript checkers in favor of eslint.

## Test instructions

This PR can be tested by following these steps:

1. Ensure you have all development dependencies for the project by running `yarn install`.
2. Execute the command `grunt check`, this is an alias that contains the following grunt commands: 
  - 'eslint'
  - 'jsonlint'
  - 'jscs'
  - 'phplint'
  - 'phpcs'
  - 'checktextdomain'
3. Verify that the new JavaScript checkers do their job.

Fixes #228

### CODE REVIEW

**Comment by moorscode** [comment]
I'm expecting the `js/src/*` files to be modified as well.
These all contain single-quoted strings and the default seems to be double-quoted strings.

Are these files included in the `eslint` list of files to be checked?

**Comment by jcomack** [comment]
CR done 👍 

Next time, please don't forget to add the proper changelog entry. I've already added it for you in regards to this PR.

**Comment by jcomack** [comment]
Acceptance failed. A few things stood out:

1. I ran the code and was instantly slapped in the face regarding CS errors. Please fix these.

2. It's odd to me that Travis isn't picking up on this. Shouldn't this be changed as well?

3. The file `.jscsrc` is still in the directory, which contains conflicting CS rules. As PhpStorm reads this file, it'll mark CS issues and thus leads to confusion.

4. The configuration file (`.eslintrc`) differs from that of `wordpress-seo`:

```
extends: yoast

parser: babel-eslint

rules:
  complexity: [1, 6]
  react/prop-types: 1
  camelcase: 1
  brace-style: 1
  max-len: 1
  max-statements: 1
  valid-jsdoc: 2
```

VS

```
{
	"parser": "babel-eslint",

	"rules":{
		"complexity": [1, 6],
		"react/prop-types": 1,
		"camelcase": 1,
		"brace-style": 1,
		"max-len": 1,
		"max-statements": 1,
		"valid-jsdoc": 2,
	}
}
```

5. Certain checks aren't being done at all. For example, in `yoastseo-woo-plugin.js` on line 81, there's an if-statement that doesn't apply the right kind of spacing inside `()` brackets. I'm uncertain whether this is due to the configuration being different (as described in point 3) or due to a missing rule. I feel that this too should be checked.

**Comment by moorscode** [comment]
Added Yoast config to make sure the base checks are configured as well.
Problems should be resolved.

Adding the checks to Travis should be done in a separate issue, after this one is done.

**Comment by dutchmartin** [comment]
Fixed all errors, some warnings are still there, but thats okey

File: Gruntfile.js [hunk 1]
```diff
@@ -1,32 +1,33 @@
 /* global require, process */
 const { flattenVersionForFile } = require( "./grunt/modules/version.js" );
+const  timeGrunt = require( "time-grunt" );
```
Comment (andizer) [inline Gruntfile.js]: There is an unnecessary space

File: js/src/yoastseo-woo-plugin.js [hunk 2]
```diff
@@ -1,55 +1,54 @@
...
 		this.registerModifications();
 
 		this.bindEvents();
 	}
 
+	/**
+	 * Adds eventlistener to load the Yoast WooCommerce plugin
+	 */
+	if( typeof YoastSEO !== "undefined" && typeof YoastSEO.app !== "undefined" ) {
+		( () => new YoastWooCommercePlugin() )();
```
Comment (andizer) [inline js/src/yoastseo-woo-plugin.js]: Is there a specific reason for rewriting the existing code to new JS standards?

File: js/src/yoastseo-woo-plugin.js [hunk 3]
```diff
@@ -61,26 +60,23 @@
...
 	};
 
 	/**
-	 * Tests the length of the productdescription.
-	 * @param {object} paper The paper to run this assessment on
-	 * @param {object} researcher The researcher used for the assessment
-	 * @param {object} i18n The i18n-object used for parsing translations
-	 * @returns {object} an assessmentresult with the score and formatted text.
+	 * Tests the length of the product description.
+	 * @returns {object} an assessment result with the score and formatted text.
```
Comment (andizer) [inline js/src/yoastseo-woo-plugin.js]: `an` should start with a capital

File: js/src/yoastseo-woo-plugin.js [hunk 4]
```diff
@@ -61,26 +60,23 @@
...
 	 */
-	YoastWooCommercePlugin.prototype.productDescription = function( paper, researcher, i18n ) {
-		var productDescription = document.getElementById( 'excerpt' ).value;
-		if (typeof tinyMCE !== 'undefined' && tinyMCE.get( 'excerpt') !== null) {
-			productDescription = tinyMCE.get( 'excerpt').getContent();
+	YoastWooCommercePlugin.prototype.productDescription = function() {
```
Comment (andizer) [inline js/src/yoastseo-woo-plugin.js]: Is it okay to remove the parameters?

File: js/src/yoastseo-woo-plugin.js [hunk 5]
```diff
@@ -93,81 +89,82 @@
...
 	 */
 	YoastWooCommercePlugin.prototype.bindEvents = function() {
-		jQuery( '.add_product_images' ).find( 'a' ).on( 'click', this.bindLinkEvent.bind( this ) );
-
+		jQuery( ".add_product_images" ).find( "a" ).on( "click", this.bindLinkEvent.bind( this ) );
 	};
 
 	/**
-	 * counters for the setTimeouts, used to make sure we don't end up in an infinite loop.
+	 * Counters for the setTimeouts, used to make sure we don"t end up in an infinite loop.
```
Comment (andizer) [inline js/src/yoastseo-woo-plugin.js]: The `"` change is not necessary.

File: js/src/yoastseo-woo-plugin.js [hunk 6]
```diff
@@ -177,40 +174,41 @@
...
-		YoastSEO.app.registerModification( 'content', callback, 'YoastWooCommercePlugin', 10 );
+		YoastSEO.app.registerModification( "content", callback, "YoastWooCommercePlugin", 10 );
 	};
 
 	/**
-	 * Adds the images from the pagegallery to the content to be analyzed by the analyzer.
-	 * @param data {String}
-	 * @returns {String}
+	 * Adds the images from the page gallery to the content to be analyzed by the analyzer.
+	 * @param {String} data The data string that has to have the images outer html concatenated on.
```
Comment (andizer) [inline js/src/yoastseo-woo-plugin.js]: 'has to have' feels weird. That should have would be better.

File: js/src/yoastseo-woo-replacevars.js [hunk 7]
```diff
@@ -57,9 +54,9 @@
 	 */
 	function extractBrandName( checkbox ) {
 		// Take the parent of checkbox with type label and clone it.
-		var clonedLabel = checkbox.parent( 'label' ).clone();
+		var clonedLabel = checkbox.parent( "label" ).clone();
 
-		// Finds child elements and removes them so we only get the label's text left.
+		// Finds child elements and removes them so we only get the label"s text left.
```
Comment (andizer) [inline js/src/yoastseo-woo-replacevars.js]: Please revert the `"` change here.

File: js/src/yoastseo-woo-plugin.js [hunk 8]
```diff
@@ -1,55 +1,54 @@
...
-		YoastSEO.app.registerPlugin( 'YoastWooCommercePlugin', { status: 'ready' } );
+		YoastSEO.app.registerPlugin( "YoastWooCommercePlugin", { status: "ready" } );
 
 		this.registerModifications();
 
 		this.bindEvents();
 	}
 
+	/**
+	 * Adds eventlistener to load the Yoast WooCommerce plugin
```
Comment (andizer) [inline js/src/yoastseo-woo-plugin.js]: It would be nice to end comments with a full stop.

File: js/src/yoastseo-woo-plugin.js [hunk 9]
```diff
@@ -1,55 +1,54 @@
...
 	 */
 	function YoastWooCommercePlugin() {
-		YoastSEO.app.registerPlugin( 'YoastWooCommerce', { 'status': 'ready' } );
+		YoastSEO.app.registerPlugin( "YoastWooCommerce", { status: "ready" } );
 
-		YoastSEO.app.registerAssessment( 'productTitle', { getResult: this.productDescription.bind( this ) }, 'YoastWooCommerce' );
+		YoastSEO.app.registerAssessment( "productTitle", { getResult: this.productDescription.bind( this ) }, "YoastWooCommerce" );
 
 		this.addCallback();
+		this.addImageToContent();
```
Comment (andizer) [inline js/src/yoastseo-woo-plugin.js]: This is a functional change? Why did you add this?

File: js/src/yoastseo-woo-replacevars.js [hunk 10]
```diff
@@ -232,38 +229,36 @@
 	 * @returns {string} The text in which the placeholders have been replaced.
 	 */
 	YoastReplaceVarPlugin.prototype.replacePlaceholders = function( text ) {
-		for ( var placeholder in placeholders ) {
-			var replaceVar = placeholders[ placeholder ];
+		for ( var i = 0; i < placeholders.length; i++ ) {
```
Comment (andizer) [inline js/src/yoastseo-woo-replacevars.js]: Why did you change the type of this loop?

File: js/src/yoastseo-woo-plugin.js (hunk 9 above)
Comment (andizer) [inline js/src/yoastseo-woo-plugin.js]: What is the reason for this functional change?

File: js/src/yoastseo-woo-plugin.js [hunk 11]
```diff
@@ -1,55 +1,54 @@
...
+	/**
+	 * Adds eventlistener to load the Yoast WooCommerce plugin
+	 */
+	if( typeof YoastSEO !== "undefined" && typeof YoastSEO.app !== "undefined" ) {
+		( () => new YoastWooCommercePlugin() )();
+	} else {
+		jQuery( window ).on(
+			"YoastSEO:ready",
+			function() {
+				( () => new YoastWooCommercePlugin() )();
```
Comment (andizer) [inline js/src/yoastseo-woo-plugin.js]: What is the reason for this change?

File: js/src/yoastseo-woo-plugin.js (hunk 2 above)
Comment (andizer) [inline js/src/yoastseo-woo-plugin.js]: Why change this into a modern js thing?

File: js/src/yoastseo-woo-plugin.js (hunk 5 above)
Comment (andizer) [inline js/src/yoastseo-woo-plugin.js]: The quote change is not necessary in comments

File: js/src/yoastseo-woo-plugin.js [hunk 12]
```diff
@@ -177,40 +174,41 @@
...
 
-		YoastSEO.app.registerModification( 'content', callback, 'YoastWooCommercePlugin', 10 );
+		YoastSEO.app.registerModification( "content", callback, "YoastWooCommercePlugin", 10 );
 	};
 
 	/**
-	 * Adds the images from the pagegallery to the content to be analyzed by the analyzer.
-	 * @param data {String}
-	 * @returns {String}
+	 * Adds the images from the page gallery to the content to be analyzed by the analyzer.
```
Comment (andizer) [inline js/src/yoastseo-woo-plugin.js]: Why change `pagegallery` to `page gallery`. `pagegallery` feels like something that is from WordPress

File: js/src/yoastseo-woo-plugin.js (hunk 6 above)
Comment (andizer) [inline js/src/yoastseo-woo-plugin.js]: This sentence feels weird.

File: js/src/yoastseo-woo-plugin.js [hunk 13]
```diff
@@ -177,40 +174,41 @@
...
+	 * @param {String} data The data string that has to have the images outer html concatenated on.
+	 * @returns {String} The data string with the images outer html concatenated on.
```
Comment (andizer) [inline js/src/yoastseo-woo-plugin.js]: Can you make this sentence more clear

File: js/src/yoastseo-woo-replacevars.js (hunk 7 above)
Comment (andizer) [inline js/src/yoastseo-woo-replacevars.js]: The quote change is not necessary

File: js/src/yoastseo-woo-plugin.js (hunk 2 above)
Comment (dutchmartin) [reply js/src/yoastseo-woo-plugin.js]: Its a way to not have eslint complain about 'new Foo()' instead of 'var foo = new Foo()', but I can disable this error too

Comment (andizer) [reply js/src/yoastseo-woo-plugin.js]: Okay, but here https://github.com/Yoast/wpseo-woocommerce/pull/236/files#diff-dd1249ecc1aea9460170266028da2f1eL263 you didn't use the same way.

Comment (dutchmartin) [reply js/src/yoastseo-woo-plugin.js]: I didn't, but I will be suppressing the error since I think this way of dealing with it is a dirty hack.

File: js/src/yoastseo-woo-plugin.js (hunk 4 above)
Comment (dutchmartin) [reply js/src/yoastseo-woo-plugin.js]: They are not used in the function (which gives a eslint error) and thats why I removed them after asking @moorscode about it.

File: js/src/yoastseo-woo-replacevars.js (hunk 7 above)
Comment (dutchmartin) [reply js/src/yoastseo-woo-replacevars.js]: eslint is not a supporter of that opinion: "Strings must use doublequote"

File: js/yoastseo-woo-plugin-590.js [hunk 14]
```diff
@@ -62,26 +61,23 @@
...
 	};
 
 	/**
-	 * Tests the length of the productdescription.
-	 * @param {object} paper The paper to run this assessment on
-	 * @param {object} researcher The researcher used for the assessment
-	 * @param {object} i18n The i18n-object used for parsing translations
-	 * @returns {object} an assessmentresult with the score and formatted text.
+	 * Tests the length of the product description.
+	 * @returns {object} an assessment result with the score and formatted text.
```
Comment (jcomack) [inline js/yoastseo-woo-plugin-590.js]: Missing whitespace above this line. Also please capitalize the `object` return type _and_ the first word in the sentence.

File: js/yoastseo-woo-plugin-590.js [hunk 15]
```diff
@@ -94,81 +90,82 @@
 	 * @returns {{score: number, text: *}} The result object with score and text.
```
Comment (jcomack) [inline js/yoastseo-woo-plugin-590.js]: Missing whitespace above this line and the one above it.

File: js/yoastseo-woo-plugin-590.js [hunk 16]
```diff
@@ -94,81 +90,82 @@
...
-				text: wpseoWooL10n.woo_desc_long
+				text: wpseoWooL10n.woo_desc_long,
 			};
 		}
 	};
 
 	/**
 	 * Adds callback to the excerpt field to trigger the analyzeTimer when product description is updated.
 	 * The tinyMCE triggers automatically since that inherets the binding from the content field tinyMCE.
+	 * @returns {void}
```
Comment (jcomack) [inline js/yoastseo-woo-plugin-590.js]: Missing whitespace above this line.

File: js/yoastseo-woo-plugin-590.js [hunk 17]
```diff
@@ -94,81 +90,82 @@
...
+		if( elem !== null ) {
+			elem.addEventListener( "input", YoastSEO.app.analyzeTimer.bind( YoastSEO.app ) );
 		}
-
 	};
 
 	/**
-	 * binds events to the add_product_images anchor.
+	 * Binds events to the add_product_images anchor.
+	 * @returns {void}
```
Comment (jcomack) [inline js/yoastseo-woo-plugin-590.js]: Missing whitespace above this line.

File: js/yoastseo-woo-plugin-590.js [hunk 18]
```diff
@@ -94,81 +90,82 @@
...
 	YoastWooCommercePlugin.prototype.bindEvents = function() {
-		jQuery( '.add_product_images' ).find( 'a' ).on( 'click', this.bindLinkEvent.bind( this ) );
-
+		jQuery( ".add_product_images" ).find( "a" ).on( "click", this.bindLinkEvent.bind( this ) );
 	};
 
 	/**
-	 * counters for the setTimeouts, used to make sure we don't end up in an infinite loop.
+	 * Counters for the setTimeouts, used to make sure we don"t end up in an infinite loop.
 	 * @type {number}
```
Comment (jcomack) [inline js/yoastseo-woo-plugin-590.js]: Missing whitespace above this line.

File: js/yoastseo-woo-plugin-590.js [hunk 19]
```diff
@@ -94,81 +90,82 @@
...
 	 */
 	YoastWooCommercePlugin.prototype.bindEvents = function() {
-		jQuery( '.add_product_images' ).find( 'a' ).on( 'click', this.bindLinkEvent.bind( this ) );
-
+		jQuery( ".add_product_images" ).find( "a" ).on( "click", this.bindLinkEvent.bind( this ) );
 	};
 
 	/**
-	 * counters for the setTimeouts, used to make sure we don't end up in an infinite loop.
+	 * Counters for the setTimeouts, used to make sure we don"t end up in an infinite loop.
```
Comment (jcomack) [inline js/yoastseo-woo-plugin-590.js]: Double-quote in comment.

File: js/yoastseo-woo-plugin-590.js [hunk 20]
```diff
@@ -94,81 +90,82 @@
...
 	 * @type {number}
 	 */
 	var buttonEventCounter = 0;
 	var deleteEventCounter = 0;
 
 	/**
-	 * after the modal dialog is opened, check for the button that adds images to the gallery to trigger
+	 * After the modal dialog is opened, check for the button that adds images to the gallery to trigger
 	 * the modification.
+	 * @returns {void}
```
Comment (jcomack) [inline js/yoastseo-woo-plugin-590.js]: Missing whitespace above this line.

File: js/yoastseo-woo-plugin-590.js [hunk 21]
```diff
@@ -94,81 +90,82 @@
...
 			buttonEventCounter = 0;
-			jQuery( '.media-modal-content' ).find( '.media-button' ).on( 'click', this.buttonCallback.bind( this )  );
+			jQuery( ".media-modal-content" ).find( ".media-button" ).on( "click", this.buttonCallback.bind( this )  );
 		}
 	};
 
 	/**
 	 * After the gallery is added, call the analyzeTimer of the app, to add the modifications.
 	 * Also call the bindDeleteEvent, to bind the analyzerTimer when an image is deleted.
+	 * @returns {void}
```
Comment (jcomack) [inline js/yoastseo-woo-plugin-590.js]: Missing whitespace above this line.

File: js/yoastseo-woo-plugin-590.js [hunk 22]
```diff
@@ -178,43 +175,44 @@
 	/**
 	 * Checks if the delete buttons of the added images are available. When they are, bind the analyzeTimer function
 	 * so when a image is removed, the modification is run.
+	 * @returns {void}
```
Comment (jcomack) [inline js/yoastseo-woo-plugin-590.js]: Missing whitespace above this line.

File: js/yoastseo-woo-plugin-590.js [hunk 23]
```diff
@@ -178,43 +175,44 @@
...
 		} else {
 			deleteEventCounter = 0;
-			jQuery( '#product_images_container' ).find( '.delete' ).on( 'click', YoastSEO.app.analyzeTimer.bind( YoastSEO.app ) );
+			jQuery( "#product_images_container" ).find( ".delete" ).on( "click", YoastSEO.app.analyzeTimer.bind( YoastSEO.app ) );
 		}
 	};
 
 	/**
 	 * Registers the addImageToContent modification
+	 * @returns {void}
```
Comment (jcomack) [inline js/yoastseo-woo-plugin-590.js]: Missing whitespace above this line.

File: js/yoastseo-woo-plugin-590.js [hunk 24]
```diff
@@ -178,43 +175,44 @@
...
 			}
 		} else {
 			deleteEventCounter = 0;
-			jQuery( '#product_images_container' ).find( '.delete' ).on( 'click', YoastSEO.app.analyzeTimer.bind( YoastSEO.app ) );
+			jQuery( "#product_images_container" ).find( ".delete" ).on( "click", YoastSEO.app.analyzeTimer.bind( YoastSEO.app ) );
 		}
 	};
 
 	/**
 	 * Registers the addImageToContent modification
```
Comment (jcomack) [inline js/yoastseo-woo-plugin-590.js]: Missing `.`.

File: js/yoastseo-woo-plugin-590.js [hunk 25]
```diff
@@ -178,43 +175,44 @@
...
-		YoastSEO.app.registerModification( 'content', callback, 'YoastWooCommercePlugin', 10 );
+		YoastSEO.app.registerModification( "content", callback, "YoastWooCommercePlugin", 10 );
 	};
 
 	/**
-	 * Adds the images from the pagegallery to the content to be analyzed by the analyzer.
-	 * @param data {String}
-	 * @returns {String}
+	 * Adds the images from the page gallery to the content to be analyzed by the analyzer.
+	 * @param {String} data The data string that has to have the images outer html concatenated on.
```
Comment (jcomack) [inline js/yoastseo-woo-plugin-590.js]: Missing whitespace above this line.

File: js/yoastseo-woo-plugin-590.js [hunk 26]
```diff
@@ -178,43 +175,44 @@
...
+	 * @returns {String} The data string with the images outer html concatenated on.
```
Comment (jcomack) [inline js/yoastseo-woo-plugin-590.js]: Missing whitespace above this line.

File: js/src/yoastseo-woo-plugin.js [hunk 27]
```diff
@@ -1,55 +1,53 @@
...
+			}
+		);
+	}
+
+	/**
+	 * Strip double spaces from text.
 	 *
 	 * @param {String} text The text to strip spaces from.
-	 * @returns {String} The text without double spaces
+	 * @returns {String} The text without double spaces.
```
Comment (jcomack) [inline js/src/yoastseo-woo-plugin.js]: Missing whitespace above this line.

Comment (jcomack) [reply js/src/yoastseo-woo-plugin.js]: Additionally, I'm unsure whether we should be using `{String}` or `{string}`. I'm tempted to say we should use the latter throughout.

File: js/src/yoastseo-woo-plugin.js [hunk 28]
```diff
@@ -61,26 +59,23 @@
 	 * @returns {String} The text without HTML-tags.
```
Comment (jcomack) [inline js/src/yoastseo-woo-plugin.js]: Additionally, I'm unsure whether we should be using `{String}` or `{string}`. I'm tempted to say we should use the latter throughout.

File: js/src/yoastseo-woo-plugin.js [hunk 29]
```diff
@@ -61,26 +59,23 @@
...
 	};
 
 	/**
-	 * Tests the length of the productdescription.
-	 * @param {object} paper The paper to run this assessment on
-	 * @param {object} researcher The researcher used for the assessment
-	 * @param {object} i18n The i18n-object used for parsing translations
-	 * @returns {object} an assessmentresult with the score and formatted text.
+	 * Tests the length of the product description.
+	 * @returns {object} An assessment result with the score and formatted text.
```
Comment (jcomack) [inline js/src/yoastseo-woo-plugin.js]: Missing whitespace above this line. Also please capitalize the word `object`.

File: js/src/yoastseo-woo-plugin.js [hunk 30]
```diff
@@ -93,81 +88,82 @@
...
 	YoastWooCommercePlugin.prototype.bindEvents = function() {
-		jQuery( '.add_product_images' ).find( 'a' ).on( 'click', this.bindLinkEvent.bind( this ) );
-
+		jQuery( ".add_product_images" ).find( "a" ).on( "click", this.bindLinkEvent.bind( this ) );
 	};
 
 	/**
-	 * counters for the setTimeouts, used to make sure we don't end up in an infinite loop.
+	 * Counters for the setTimeouts, used to make sure we don't end up in an infinite loop.
 	 * @type {number}
```
Comment (jcomack) [inline js/src/yoastseo-woo-plugin.js]: Missing whitespace above this line.

File: js/src/yoastseo-woo-plugin.js [hunk 31]
```diff
@@ -177,40 +173,41 @@
...
-		YoastSEO.app.registerModification( 'content', callback, 'YoastWooCommercePlugin', 10 );
+		YoastSEO.app.registerModification( "content", callback, "YoastWooCommercePlugin", 10 );
 	};
 
 	/**
-	 * Adds the images from the pagegallery to the content to be analyzed by the analyzer.
-	 * @param data {String}
-	 * @returns {String}
+	 * Adds the images from the page gallery to the content to be analyzed by the analyzer.
+	 * @param {String} data The data string that does not have the images outer html.
```
Comment (jcomack) [inline js/src/yoastseo-woo-plugin.js]: Missing whitespace above this line.

Additionally, I'm unsure whether we should be using `{String}` or `{string}`. I'm tempted to say we should use the latter throughout.

File: js/src/yoastseo-woo-plugin.js [hunk 32]
```diff
@@ -177,40 +173,41 @@
...
+	 * @returns {String} The data string parameter with the images outer html.
```
Comment (jcomack) [inline js/src/yoastseo-woo-plugin.js]: Missing whitespace above this line.

Additionally, I'm unsure whether we should be using `{String}` or `{string}`. I'm tempted to say we should use the latter throughout.

File: js/yoastseo-woo-plugin-590.js (hunk 25 above)
Comment (jcomack) [reply js/yoastseo-woo-plugin-590.js]: I'm unsure whether we should be using `{String}` or `{string}`. I'm tempted to say we should use the latter throughout.

File: js/yoastseo-woo-plugin-590.js (hunk 26 above)
Comment (jcomack) [reply js/yoastseo-woo-plugin-590.js]: I'm unsure whether we should be using `{String}` or `{string}`. I'm tempted to say we should use the latter throughout.

File: js/src/yoastseo-woo-plugin.js (hunk 32 above)
Comment (dutchmartin) [reply js/src/yoastseo-woo-plugin.js]: jsdoc uses lowercase: http://usejsdoc.org/about-getting-started.html

Comment (dutchmartin) [reply js/src/yoastseo-woo-plugin.js]: Since the uppercase "String" is consistent, I will leave it as it is.

Comment (jcomack) [reply js/src/yoastseo-woo-plugin.js]: You mention that jsdoc (which is the official standard) uses lowercase. Why not change it then?

**Comment by jcomack** [comment]
Build still isn't working and there's a conflict in a language file. Can you please resolve these issues? Additionally, there's still some unprocessed feedback from a previous CR.

**Comment by dutchmartin** [comment]
Travis now succeeds, and the merge is completed. But I cannot solve the codeclimate issues since I have no access. I guess code climate complains about 4 `new Foo()` object creations without variables to store them like `var Bar = new Bar()`

File: js/src/yoastseo-woo-replacevars.js (hunk 7 above)
Comment (moorscode) [reply js/src/yoastseo-woo-replacevars.js]: But this is not a textually correct sentence. It needs to use a single quote.
This is not a string, this is a comment.

File: js/src/yoastseo-woo-replacevars.js (hunk 10 above)
Comment (moorscode) [reply js/src/yoastseo-woo-replacevars.js]: Please revert this change, as it should not be done it this PR, if done at all.
